pkg/config/data_test.go

```diff
@@ -36,3 +45,276 @@ func TestConfigLoad(t *testing.T) {
+			// When we're done, chdir back to our original location.
+			// This is so we can reliably assert file structure.
+			if err := os.Chdir(rootdir); err != nil {
+				t.Fatal(err)
+			}
+			defer os.Chdir(pwd)
+
+			if testcase.userConfigFilename == "" {
+				if fi, err := os.Stat(fpath); err == nil {
+					t.Fatalf("expected the user config to NOT exist at this point: %+v", fi)
+				}
+			} else {
+				if _, err := os.Stat(fpath); err != nil {
+					t.Fatalf("expected the user config to exist at this point: %v", err)
+				}
+			}
+
+			var f config.File
+			f.Static = testcase.staticConfig
+
+			var out bytes.Buffer
+			in := strings.NewReader(testcase.userResponseToPrompt)
+
+			err = f.Read(fpath, in, &out)
+
+			if testcase.remediation {
+				e, ok := err.(fsterr.RemediationError)
+				if !ok {
+					t.Fatalf("unexpected error asserting returned error (%T) to a RemediationError type", err)
+				}
+				if testcase.wantError != e.Remediation {
+					t.Fatalf("want %v, have %v", testcase.wantError, e.Remediation)
+				}
+			} else {
+				testutil.AssertErrorContains(t, err, testcase.wantError)
+			}
+
+			if testcase.wantError == "" {
+				// If we're not expecting an error, then we're expecting the user
+				// configuration file to exist...
+
+				if _, err := os.Stat(fpath); err == nil {
+					bs, err := os.ReadFile(fpath)
+					if err != nil {
+						t.Fatalf("unexpected err: %v", err)
+					}
+
+					err = toml.Unmarshal(bs, &f)
+					if err != nil {
+						t.Fatalf("unexpected err: %v", err)
+					}
+
+					if f.CLI.LastChecked == "" || f.CLI.Version == "" {
+						t.Fatalf("expected LastChecked/Version to be set: %+v", f)
+					}
+				}
+			}
+		})
+	}
+}
+
+// TestUseStatic validates all of the logic flow within config.File.UseStatic()
+func TestUseStatic(t *testing.T) {
+	// We're going to chdir to an temp environment,
+	// so save the PWD to return to, afterwards.
+	pwd, err := os.Getwd()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Create our environment in a temp dir.
+	// Defer a call to clean it up.
+	rootdir, fpath := makeTempEnvironment(t, "config-legacy.toml")
+	defer os.RemoveAll(rootdir)
+
+	// Before running the test, chdir into the temp environment.
+	// When we're done, chdir back to our original location.
+	// This is so we can reliably assert file structure.
+	if err := os.Chdir(rootdir); err != nil {
+		t.Fatal(err)
+	}
+	defer os.Chdir(pwd)
+
+	f := config.File{}
+
+	// Validate that invalid static configuration returns a specific error.
+	err = f.UseStatic(staticConfigInvalid, fpath)
+	if err == nil {
+		t.Fatal("expected an error, but got nil")
+	} else {
+		testutil.AssertErrorContains(t, err, config.ErrInvalidConfig.Error())
+	}
+
+	// Validate that legacy configuration can be migrated to the static one
+	// embedded in the CLI binary.
+	var out bytes.Buffer
+	f.Read(fpath, strings.NewReader(""), &out)
+	f.UseStatic(staticConfig, fpath)
+	if f.CLI.LastChecked == "" || f.CLI.Version == "" {
```

Comment:
Ensure we're able to validate that we have modern config that needs updating to latest version and so we validate the tokens aren't lost.